File: app/src/main/java/wmsdf/cl/exp4/getwallpaper/intent/PhotoSelectContract.java
package wmsdf.cl.exp4.getwallpaper.intent;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import androidx.activity.result.contract.ActivityResultContract;

public class PhotoSelectContract extends ActivityResultContract<java.lang.Integer, Uri> {

    @Override
    public Intent createIntent(Context context, java.lang.Integer input) {
        return new Intent(Intent.ACTION_PICK).setType("image/*");
    }

    @Override
    public Uri parseResult(int resultCode, Intent intent) {
        if(intent == null) {
            return null;
        }
        return intent.getData();
    }

}

File: kotlin/src/main/java/wmsdf/cl/exp4/getwallpaper/kotlin/intent/PhotoSelectContract.kt
package wmsdf.cl.exp4.getwallpaper.kotlin.intent

import android.content.Context
import android.content.Intent
import android.net.Uri
import androidx.activity.result.contract.ActivityResultContract

class PhotoSelectContract : ActivityResultContract<Unit, Uri?>() {

    override fun createIntent(context: Context, input: Unit): Intent {
        return Intent(Intent.ACTION_PICK).setType("image/*")
    }

    override fun parseResult(resultCode: Int, intent: Intent?): Uri? {
        if(intent == null) {
            return null
        }
        return intent.data
    }

}